zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FlowNodeInstanceFromIncidentHandler.java

```diff
@@ -41,13 +44,18 @@ public Class<FlowNodeInstanceEntity> getEntityType() {
 
   @Override
   public boolean handlesRecord(final Record<IncidentRecordValue> record) {
-    return true;
+    final String intentStr = record.getIntent().name();
+    return intentStr.equals(IncidentIntent.CREATED.toString())
+        || intentStr.equals(IncidentIntent.MIGRATED.toString());
   }
 
   @Override
   public List<String> generateIds(final Record<IncidentRecordValue> record) {
+    // in case of incident we update tree path in all flow node instances in call hierarchy
     final IncidentRecordValue recordValue = record.getValue();
-    return List.of(String.valueOf(recordValue.getElementInstanceKey()));
+    final List<Long> callHierarchy = recordValue.getElementInstancePath().getLast();
```

Comment:
❓ Why we get the last?

Reply:
The last element in `getElementInstancePath` is about the process instance on which incident was created. This is the instance where we want to set tree paths for flow node instances.